framework/audio/engine/iaudiodriver.h

```diff
@@ -0,0 +1,69 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_IAUDIODRIVER_H
+#define MU_AUDIO_IAUDIODRIVER_H
+
+#include <cstdint>
+#include <string>
+#include <functional>
+
+#include "modularity/imoduleexport.h"
+
+namespace mu {
+namespace audio {
+namespace engine {
+
+class IAudioDriver : MODULE_EXPORT_INTERFACE
+{
+    INTERFACE_ID(IAudioDriver)
+
+public:
+    virtual ~IAudioDriver() = default;
+
+    using DeviceID = uint32_t;
```

Comment:
Maybe it'll be better to use a structure with the isValid method

Reply:
I think this is redundant

Reply:
I removed `DeviceID` because at the moment it is still not supported.